Objective-C/CBLMutableDocument.mm

```diff
@@ -183,9 +183,10 @@ - (C4SliceResult) encode: (NSError**)outError {
         return {};
     }
     FLError flErr;
+    const char* errMessage = FLEncoder_GetErrorMessage(encoder);
     FLSliceResult body = FLEncoder_Finish(encoder, &flErr);
     if (!body.buf)
```

Comment:
Shouldn't you look up the error message only if you need it?

Reply:
there is an issue with it, we knew if there is any error once the `FLEncoder_Finish` function finishes(from FLError), and once it is finish, it will reset the encoder, so the previous recorded error message will be empty.